pkg/controller/installation/core_controller.go

```diff
@@ -931,12 +935,37 @@ func (r *ReconcileInstallation) Reconcile(ctx context.Context, request reconcile
 
 	var typhaNodeTLS *render.TyphaNodeTLS
 	if instance.Spec.CertificateManagement == nil {
+		// First, attempt to load TLS secrets from the cluster, if any exist.
```

Comment:
I need to revisit this since I don't think I fully understand how it was working before, but it seemed like we had a strange split of TLS secret generation between the controller and the render package.

The approach I'm trying to move towards would to have the controller be responsible for generating the secrets when needed, and caching them in its own namespace. It can pass them as inputs to the render package, which can then modify / move them to the correct destination namespace and return them for installation into the cluster.

Reply:
Agreed, I think this is where we decided to go after that operator meeting like 2 months ago.